source/application/wintaser.cpp

```diff
@@ -2885,177 +2874,79 @@ LPCWSTR ExceptionCodeToDescription(DWORD code)
-	while(VirtualQueryEx(hProcess, (char*)hModule+size+0x1000, &mbi, sizeof(mbi))
-	&& (DWORD)mbi.AllocationBase == (DWORD)hModule)
-		size += mbi.RegionSize;
-	return std::max<DWORD>(size, 0x10000);
+    /*
+     * If we can "trust" a newly loaded module, add it to the list.
+     */
+    int proximity = 2;
+    if (!wcsicmp(path.c_str(), injected_dll_path.c_str()))
+    {
+        trusted_modules[process_id].emplace_back(path);
+    }
+    else if (CountBackslashes(exe_filename.c_str() + CountSharedPrefixLength(exe_filename, path)) < proximity)
+    {
+        trusted_modules[process_id].emplace_back(path);
+    }
+    else if (CountBackslashes(sub_exe_filename.c_str() + CountSharedPrefixLength(sub_exe_filename, path)) < proximity)
+    {
+        trusted_modules[process_id].emplace_back(path);
+    }
+    else if (path.length() >= 4 && !wcsnicmp(path.c_str() + path.length() - 4, L".cox", 4)) // hack, we can generally assume the game outputted any dll that has this extension
+    {
+        trusted_modules[process_id].emplace_back(path);
+    }
 }
 
-void RegisterModuleInfo(LPVOID hModule, HANDLE hProcess, LPCWSTR path)
+static void ModuleUnloaded(DWORD process_id, const std::wstring& path)
 {
-	int trusted = IsPathTrusted(path);
-	if(!trusted)
-		return; // we only want to keep track of the trusted ones
+    auto& proc_modules = trusted_modules[process_id];
+    auto removed = std::remove_if(proc_modules.begin(), proc_modules.end(),
+                                  [&path](const std::wstring& m) { return (m.compare(path) == 0); });
 
-	MyMODULEINFO mmi;
-//	if(!GetModuleInformation(hProcess, (HMODULE)hModule, &mmi.mi, sizeof(mmi.mi)))
-	{
-		// GetModuleInformation never works, and neither does EnumProcessModules.
-		// I think it's because we're calling it too early.
-		// (either that or some versions of PSAPI.DLL are simply busted)
-		// so this is the best fallback I could come up with
-		mmi.mi.lpBaseOfDll = hModule;
-		mmi.mi.SizeOfImage = CalculateModuleSize(hModule, hProcess);
-		mmi.mi.EntryPoint = hModule; // don't care
-	}
-	mmi.path = path;
-	if(trusted == 2)
-	{
-		injectedDllModuleInfo = mmi;
-		//SendTrustedRangeInfos(hProcess); // disabled because it's probably not ready to receive it yet.
-	}
-	else
-	{
-		for(unsigned int i = 0; i < trustedModuleInfos.size(); i++)
-		{
-			MyMODULEINFO& mmi2 = trustedModuleInfos[i];
-			if((DWORD)mmi.mi.lpBaseOfDll >= (DWORD)mmi2.mi.lpBaseOfDll
-			&& (DWORD)mmi.mi.lpBaseOfDll+mmi.mi.SizeOfImage <= (DWORD)mmi2.mi.lpBaseOfDll+mmi2.mi.SizeOfImage)
-			{
-				debugprintf(L"apparently already TRUSTED MODULE 0x%08X - 0x%08X (%s)\n", mmi.mi.lpBaseOfDll, (DWORD)mmi.mi.lpBaseOfDll+mmi.mi.SizeOfImage, path);
-				return;
-			}
-		}
-		trustedModuleInfos.push_back(mmi);
-		SendTrustedRangeInfos(hProcess);
-	}
-	debugprintf(L"TRUSTED MODULE 0x%08X - 0x%08X (%s)\n", mmi.mi.lpBaseOfDll, (DWORD)mmi.mi.lpBaseOfDll+mmi.mi.SizeOfImage, path);
+    if (removed == proc_modules.end())

```

Comment:
I'm pretty sure this check isn't needed, `.erase()` will work correctly regardless.